language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/inlayhint/InlayHintProvider.java

```diff
@@ -128,12 +144,68 @@ private static Pair<List<ParameterSymbol>, Optional<ParameterSymbol>> getParamet
             // Since the method is a lang-lib function, skip the first parameter
             return Pair.of(libFunction.get().params().get().stream().skip(1).collect(Collectors.toList()),
                     libFunction.get().restParam());
+        } else if (invokableNode.kind() == SyntaxKind.CLIENT_RESOURCE_ACCESS_ACTION) {
+            return context.currentSemanticModel().get()
+                    .symbol(invokableNode)
+                    .map(symbol -> ((ResourceMethodSymbol) symbol).typeDescriptor())
+                    .map(typeSymbol -> Pair.of(typeSymbol.params().orElse(Collections.emptyList()),
+                            typeSymbol.restParam()))
+                    .orElse(Pair.of(Collections.emptyList(), Optional.empty()));
+        } else if (invokableNode.kind() == SyntaxKind.REMOTE_METHOD_CALL_ACTION) {
+            return context.currentSemanticModel().get()
+                    .symbol(invokableNode)
+                    .map(symbol -> ((MethodSymbol) symbol).typeDescriptor())
+                    .map(typeSymbol -> Pair.of(typeSymbol.params().orElse(Collections.emptyList()),
+                            typeSymbol.restParam()))
+                    .orElse(Pair.of(Collections.emptyList(), Optional.empty()));
+        } else if (invokableNode.kind() == SyntaxKind.IMPLICIT_NEW_EXPRESSION) {
+            return getParameterSymbolsForImplicitExpressions(context, invokableNode);
+        } else if (invokableNode.kind() == SyntaxKind.EXPLICIT_NEW_EXPRESSION) {
+            return getParameterSymbolsForExplicitExpressions(context, invokableNode);
         } else {
             FunctionCallExpressionNode functionCallExpressionNode = (FunctionCallExpressionNode) invokableNode;
             return getParameterSymbolsForFunctionCall(context, functionCallExpressionNode);
         }
     }
 
+    private static Pair<List<ParameterSymbol>, Optional<ParameterSymbol>> getParameterSymbolsForImplicitExpressions(
+            InlayHintContext context,
+            NonTerminalNode node) {
+        Optional<TypeSymbol> symbol = context.currentSemanticModel()
+                .flatMap(semanticModel -> semanticModel.typeOf(node))
+                .flatMap(typeSymbol -> Optional.of(CommonUtil.getRawType(typeSymbol))).stream().findFirst();
+        return getNewExpressionSymbols(symbol);
+    }
+
+    private static Pair<List<ParameterSymbol>, Optional<ParameterSymbol>> getParameterSymbolsForExplicitExpressions(
+            InlayHintContext context,
+            NonTerminalNode node) {
+        Optional<TypeSymbol> symbol = context.currentSemanticModel()
+                .flatMap(semanticModel -> semanticModel.typeOf(node))
+                .flatMap(typeSymbol -> Optional.of(CommonUtil.getRawType(typeSymbol))).stream().findFirst();
+        return getNewExpressionSymbols(symbol);
+    }
+
+    private static Pair<List<ParameterSymbol>, Optional<ParameterSymbol>> getNewExpressionSymbols(
```

Comment:
Shall we rename the method to `getParametersOfNewExpression`, and let's remove the optional parameter.